flink-table/flink-table-common/src/main/java/org/apache/flink/table/sink/filesystem/FileSystemFileCommitter.java

```diff
@@ -0,0 +1,264 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.sink.filesystem;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.core.fs.FileStatus;
+import org.apache.flink.core.fs.FileSystem;
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.table.api.TableException;
+import org.apache.flink.util.Preconditions;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Default file system {@link FileCommitter} implementation. It just move all files to
+ * output path from temporary path.
+ *
+ * <p>NOTE: Here, we have no way to update catalog.
```

Comment:
So when we would update catalog? The result of "show partitions" should be changed after we commit output files?

Reply:
Hive has metastore api to access metastore to update catalog.
But non-hive catalog not have api to update catalog in current callback. So show partitions not work. We can only read and write partitions from paths.

I think the final solution is the client refactor, provides the way to update catalog after job finish or checkpoint finish.

Reply:
But I think we can provided interface to update catalog, when it is in hive catalog, it can update catalog. What do you think?